languages/C.groovy

```diff
@@ -0,0 +1,379 @@
+@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
+import com.ibm.dbb.metadata.*
+import com.ibm.dbb.dependency.*
+import com.ibm.dbb.build.*
+import groovy.transform.*
+import com.ibm.jzos.ZFile
+import com.ibm.dbb.build.report.*
+import com.ibm.dbb.build.report.records.*
+
+
+// define script properties
+@Field BuildProperties props = BuildProperties.getInstance()
+@Field def buildUtils= loadScript(new File("${props.zAppBuildDir}/utilities/BuildUtilities.groovy"))
+@Field def impactUtils= loadScript(new File("${props.zAppBuildDir}/utilities/ImpactUtilities.groovy"))
+@Field def bindUtils= loadScript(new File("${props.zAppBuildDir}/utilities/BindUtilities.groovy"))
+@Field def resolverUtils = loadScript(new File("${props.zAppBuildDir}/utilities/ResolverUtilities.groovy"))
+
+println("** Building files mapped to ${this.class.getName()}.groovy script")
+
+// verify required build properties
+buildUtils.assertBuildProperties(props.cc_requiredBuildProperties)
+
+// create language datasets
+def langQualifier = "cc"
+buildUtils.createLanguageDatasets(langQualifier)
+
+// sort the build list based on build file rank if provided
+List<String> sortedList = buildUtils.sortBuildList(argMap.buildList, 'cc_fileBuildRank')
+
+if (buildListContainsTests(sortedList)) {
+    langQualifier = "cc_test"
+    buildUtils.createLanguageDatasets(langQualifier)
+}
+
+// iterate through build list
+sortedList.each { buildFile ->
+    println "*** Building file $buildFile"
```

Comment:
Can you please add the logic for the [Progress Indicator](https://github.com/IBM/dbb-zappbuild/blob/main/languages/Cobol.groovy#L37), that is part of the other language scripts?